sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -31,14 +31,20 @@ import org.apache.spark.sql.types.{LongType, StringType, StructType}
 
 class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAndAfter {
```

Comment:
Now this end-to-end test checks the actual behavior the end-users will hit, as we don't set a fake session catalog.